google/cloud/bigquery/client.py

```diff
@@ -471,9 +471,20 @@ def create_dataset(
                 retry, method="POST", path=path, data=data, timeout=timeout
             )
             return Dataset.from_api_repr(api_response)
-        except google.api_core.exceptions.Conflict:
+        except google.api_core.exceptions.Conflict as exc:
             if not exists_ok:
-                raise
+                if "already exists" in exc.message.lower():
+                    rebranded_error = google.api_core.exceptions.AlreadyExists(
+                        exc.message, exc.errors, exc.response
+                    )
+                    six.raise_from(rebranded_error, exc)
+                elif "aborted" in exc.message.lower():
+                    rebranded_error = google.api_core.exceptions.Aborted(
+                        exc.message, exc.errors, exc.response
+                    )
+                    six.raise_from(rebranded_error, exc)
+                else:
+                    raise
```

Comment:
Do we want to have this entire block indented under the `if not exists_ok` check, or would it make more sense to just consider that flag for the "already exists" errors?

(applies to all occurrences)